src/Symfony/Component/Mime/Email.php

```diff
@@ -423,12 +423,12 @@ public function getBody(): AbstractPart
      */
     private function generateBody(): AbstractPart
     {
-        if (null === $this->text && null === $this->html) {
-            throw new LogicException('A message must have a text and/or an HTML part.');
+        [$htmlPart, $attachmentParts, $inlineParts] = $this->prepareParts();
+        if (null === $this->text && null === $this->html && empty($attachmentParts)) {
```

Comment:
We are trying to avoid using `empty` is Symfony. `&& !$attachmentParts` should work the same.

Reply:
Done